channeldb/payment_control_test.go

```diff
@@ -196,6 +197,155 @@ func TestPaymentControlSwitchFail(t *testing.T) {
 	}
 }
 
+// TestPaymentControlSwitchFailDeletePayment checks that payment status returns to Failed
+// status after failing, and that InitPayment allows another HTLC for the
+// same payment hash.
+func TestPaymentControlSwitchFailDeletePayment(t *testing.T) {
+	t.Parallel()
+
+	db, cleanup, err := MakeTestDB()
+	defer cleanup()
+	if err != nil {
+		t.Fatalf("unable to init db: %v", err)
+	}
+
+	pControl := NewPaymentControl(db)
+
+	info, attempt, preimg, err := genInfo()
+	if err != nil {
+		t.Fatalf("unable to generate htlc message: %v", err)
+	}
+
+	// Sends base htlc message which initiate StatusInFlight.
+	err = pControl.InitPayment(info.PaymentIdentifier, info)
+	if err != nil {
+		t.Fatalf("unable to send htlc message: %v", err)
+	}
+
+	assertPaymentIndex(t, pControl, info.PaymentIdentifier)
+	assertPaymentStatus(t, pControl, info.PaymentIdentifier, StatusInFlight)
+	assertPaymentInfo(
+		t, pControl, info.PaymentIdentifier, info, nil, nil,
+	)
+
+	// Lookup the payment so we can get its old sequence number before it is
+	// overwritten.
+	payment, err := pControl.FetchPayment(info.PaymentIdentifier)
+	assert.NoError(t, err)
```

Comment:
nit: use `require` instead

Reply:
Done.

I used `assert.NoError` here because another test in this file also was. There are a handful of `assert.NoError` instances that must have slipped through in the past. I added a commit to this PR that changes them.

Reply:
Yeah I saw that one, very nice PR btw! We did have a mixed usage of both, and decided to stick to `require` but never had the time to clean the old ones, so thanks for that PR!